src/app/static/src/js/components/runWorkflow/runWorkflowCreate.vue

```diff
@@ -1,44 +1,182 @@
+                    :disabled="enableButtons">Clone workflow
+            </button>
+        </div>
+        <div class="pt-4 col-sm-6">
+            <error-info :default-message="defaultMessage" :api-error="error"></error-info>
         </div>
     </div>
 </template>
 
 <script lang="ts">
-import Vue from "vue"
+    import Vue from "vue"
+    import vSelect from 'vue-select'
+    import {api} from "../../utils/api";
+    import {RunWorkflowMetadata} from "../../utils/types";
+    import {longTimestamp} from "../../utils/helpers";
+    import ErrorInfo from "../errorInfo.vue";
+
+    interface Methods {
+        create: () => void
+        clone: () => void
+        rerun: () => void
+        getWorkflows: () => void
+        getWorkflowDetails: () => void
+        searchWorkflows: (options: [], search: string) => void
+        getLongTimestamp: (date: string) => string
+    }
+
+    interface Data {
+        workflows: []
+        error: string | null
+        defaultMessage: string | null
+        selectedWorkflow: string
+        runWorkflowMetadata: RunWorkflowMetadata | null
+    }
 
-interface Methods {
-    create: () => void
-    clone: () => void
-    rerun: () => void
-}
+    let emptyWorkflowMetadata = {
+        name: "",
+        date: "",
+        email: "",
+        reports: [],
+        instances: {},
+        git_branch: null,
+        git_commit: null,
+        key: ""
+    }
+
+    interface Computed {
+        enableButtons: boolean
+    }
 
-export default Vue.extend<unknown, Methods, unknown, unknown>({
-    name: "runWorkflowCreate",
-    methods: {
-        create: function () {
-            this.$emit("create")
+    export default Vue.extend<Data, Methods, Computed, unknown>({
+        name: "runWorkflowCreate",
+        data(): Data {
+            return {
+                workflows: [],
+                error: "",
+                defaultMessage: "",
+                selectedWorkflow: null,
+                runWorkflowMetadata: null
+            }
+        },
+        computed: {
+            enableButtons: function () {
+                return !this.selectedWorkflow || !this.runWorkflowMetadata
+            }
         },
-        clone: function () {
-            this.$emit("clone")
+        methods: {
+            create: function () {
+                this.$emit("create")
+            },
+            clone: function () {
+                if (this.selectedWorkflow && this.runWorkflowMetadata) {
+                    const {reports, git_branch, git_commit} = this.runWorkflowMetadata
+                    const clonedWorkflow: RunWorkflowMetadata = {
+                        ...emptyWorkflowMetadata,
+                        reports,
+                        git_branch,
+                        git_commit
+                    }
+                    this.$emit("clone", clonedWorkflow)
+                }
+            },
+            rerun: function () {
+                if (this.selectedWorkflow) {
```

Comment:
- Any reason why this is not identical to line 107?
- Do we need these checks at all, given the buttons should be disabled anyway?